tests/test_ext_autodoc_configs.py

```diff
@@ -1142,6 +1142,94 @@ def test_autodoc_typehints_description_and_type_aliases(app):
             '      myint\n' == context)
 
 
+@pytest.mark.sphinx('html', testroot='ext-autodoc',
+                    confoverrides={'autodoc_unqualified_typehints': True})
+def test_autodoc_unqualified_typehints(app):
+    options = {"members": None,
+               "undoc-members": None}
+    actual = do_autodoc(app, 'module', 'target.typehints', options)
+    assert list(actual) == [
+        '',
+        '.. py:module:: target.typehints',
+        '',
+        '',
+        '.. py:data:: CONST1',
+        '   :module: target.typehints',
+        '   :type: int',
+        '',
+        '',
+        '.. py:class:: Math(s: str, o: ~typing.Optional[~typing.Any] = None)',
```

Comment:
Internally, autodoc generates a typehints leading `~` if the option is enabled. This is a special notation to suppress prefix (modname and leading class name) of the typehint. It's like the `~` of `:py:class:` role (ex. ```:py:class:`~io.StringIO` ```).